OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -134,6 +135,7 @@ public Armament(Actor self, ArmamentInfo info)
 				barrels.Add(new Barrel { Offset = WVec.Zero, Yaw = WAngle.Zero });
 
 			Barrels = barrels.ToArray();
+			attack = self.TraitsImplementing<AttackBase>().FirstOrDefault(a => a.Info.Armaments.Contains(Info.Name));
```

Comment:
I wanted to avoid caching all attack traits, but you're right, if the 'right' Attack* trait gets disabled and a different one enabled this could result in wrong behavior. I'll change it.